File: AmazonParcelDelivery.cpp
#include <bits/stdc++.h>

using namespace std;

int getMinimumCost(vector<int> parcels, int k)
{
    int ans = 0;
    int count = k - parcels.size();
    unordered_map<long, long> hsh;
    for (size_t i = 0; i < count; i++)
    {
        hsh[parcels[i]]++;
    }
    for (size_t i = 1; i <= k; i++)
    {
        if (count == 0)
        {
            break;
        }
        if (hsh[i] == 0)
        {
            ans += i;
            count -= 1;
        }
    }
    return ans;
}

int main()
{
    int n, m;
    cin >> n;
    vector<int> processes;

    for (size_t i = 0; i < n; i++)
    {
        int p;
        cin >> p;
        processes.push_back(p);
    }
    cin >> m;
    int v = getMinimumCost(processes, m);
    cout << v;
    return 0;
}

File: AmazonParcelDelivery.py
def getMinimumCost(parcels, k):
    # Write your code here
    ans = 0
    count = k-len(parcels)
    arr = [len(parcels)]
    for i in range(1, k+1):
        arr[parcels[i-1]] = 1
    print("R")
    for i in range(1, k):
        if(count == 0):
            break
        # if(i not in parcels):

        if(arr[i] == 0):
            ans += i
            count -= 1

    return ans


print(getMinimumCost([4, 5, 7, 1], 4))
print(getMinimumCost([6, 5, 4, 1, 3], 7))
